File: src/test/java/com/flexpoker/test/util/datageneration/CardGenerator.java
package com.flexpoker.test.util.datageneration;

import com.flexpoker.model.card.Card;
import com.flexpoker.model.card.CardRank;
import com.flexpoker.model.card.CardSuit;
import com.flexpoker.model.card.FlopCards;
import com.flexpoker.model.card.PocketCards;
import com.flexpoker.model.card.RiverCard;
import com.flexpoker.model.card.TurnCard;

public class CardGenerator {

    private static final Card CARD_1 = new Card(1, CardRank.ACE, CardSuit.CLUBS);

    private static final Card CARD_2 = new Card(2, CardRank.TWO, CardSuit.CLUBS);

    private static final Card CARD_3 = new Card(3, CardRank.THREE, CardSuit.CLUBS);

    private static final Card CARD_4 = new Card(4, CardRank.FOUR, CardSuit.CLUBS);

    private static final Card CARD_5 = new Card(5, CardRank.FIVE, CardSuit.CLUBS);

    private static final Card CARD_6 = new Card(6, CardRank.SIX, CardSuit.CLUBS);

    private static final Card CARD_7 = new Card(7, CardRank.SEVEN, CardSuit.CLUBS);

    private static final Card CARD_8 = new Card(8, CardRank.EIGHT, CardSuit.CLUBS);

    private static final Card CARD_9 = new Card(9, CardRank.NINE, CardSuit.CLUBS);

    private static final PocketCards pocketCards1 = new PocketCards(CARD_6, CARD_7);

    private static final PocketCards pocketCards2 = new PocketCards(CARD_8, CARD_9);

    public static FlopCards createFlopCards() {
        return new FlopCards(CARD_1, CARD_2, CARD_3);
    }

    public static TurnCard createTurnCard() {
        return new TurnCard(CARD_4);
    }

    public static RiverCard createRiverCard() {
        return new RiverCard(CARD_5);
    }

    public static PocketCards createPocketCards1() {
        return pocketCards1;
    }

    public static PocketCards createPocketCards2() {
        return pocketCards2;
    }

}

File: src/test/kotlin/com/flexpoker/test/util/datageneration/CardGenerator.kt
package com.flexpoker.test.util.datageneration

import com.flexpoker.table.command.Card
import com.flexpoker.table.command.CardRank
import com.flexpoker.table.command.CardSuit
import com.flexpoker.table.command.FlopCards
import com.flexpoker.table.command.PocketCards
import com.flexpoker.table.command.RiverCard
import com.flexpoker.table.command.TurnCard

object CardGenerator {

    private val CARD_1 = Card(1, CardRank.ACE, CardSuit.CLUBS)
    private val CARD_2 = Card(2, CardRank.TWO, CardSuit.CLUBS)
    private val CARD_3 = Card(3, CardRank.THREE, CardSuit.CLUBS)
    private val CARD_4 = Card(4, CardRank.FOUR, CardSuit.CLUBS)
    private val CARD_5 = Card(5, CardRank.FIVE, CardSuit.CLUBS)
    private val CARD_6 = Card(6, CardRank.SIX, CardSuit.CLUBS)
    private val CARD_7 = Card(7, CardRank.SEVEN, CardSuit.CLUBS)
    private val CARD_8 = Card(8, CardRank.EIGHT, CardSuit.CLUBS)
    private val CARD_9 = Card(9, CardRank.NINE, CardSuit.CLUBS)
    private val pocketCards1 = PocketCards(CARD_6, CARD_7)
    private val pocketCards2 = PocketCards(CARD_8, CARD_9)

    fun createFlopCards(): FlopCards {
        return FlopCards(CARD_1, CARD_2, CARD_3)
    }

    fun createTurnCard(): TurnCard {
        return TurnCard(CARD_4)
    }

    fun createRiverCard(): RiverCard {
        return RiverCard(CARD_5)
    }

    fun createPocketCards1(): PocketCards {
        return pocketCards1
    }

    fun createPocketCards2(): PocketCards {
        return pocketCards2
    }

}